src/StoreFileSystem.cc

```diff
@@ -70,6 +70,17 @@ StoreFileSystem::FreeAllFs()
     }
 }
 
+StoreFileSystem *
+StoreFileSystem::FindByType(const char *type)
+{
+    for (size_t i = 0; i < FileSystems().size(); ++i) {
+        auto fs = FileSystems().at(i);
```

Comment:
I hope that something like that would work instead of calling FileSystems() so many times:

```suggestion
    for (const auto fs: FileSystems()) {
```